respokeSDK/src/main/java/com/digium/respokesdk/RespokeClient.java

```diff
@@ -513,6 +535,188 @@ public RespokeGroup getGroup(String groupIDToFind) {
+                    "Please reconnect!");
+            return;
+        }
+
+        if ((maxMessages == null) || (maxMessages < 1)) {
+            getGroupHistoriesError(completionListener, "maxMessages must be at least 1");
+            return;
+        }
+
+        if ((groupIds == null) || (groupIds.size() == 0)) {
+            getGroupHistoriesError(completionListener, "At least 1 group must be specified");
+            return;
+        }
+
+        Uri.Builder builder = new Uri.Builder();
+        builder.appendQueryParameter("limit", maxMessages.toString());
+        for (String id: groupIds) {
+            builder.appendQueryParameter("groupIds", id);
+        }
+
+        String urlEndpoint = "/v1/group-histories" + builder.build().toString();
+        signalingChannel.sendRESTMessage("get", urlEndpoint, null,
+                new RespokeSignalingChannel.RESTListener() {
+            @Override
+            public void onSuccess(Object response) {
+                if (!(response instanceof JSONObject)) {
+                    getGroupHistoriesError(completionListener, "Invalid response from server");
+                    return;
+                }
+
+                final JSONObject json = (JSONObject) response;
+                final HashMap<String, List<RespokeGroupMessage>> results = new HashMap<>();
+
+                for (Iterator<String> keys = json.keys(); keys.hasNext();) {
+                    final String key = keys.next();
+
+                    try {
+                        final JSONArray jsonMessages = json.getJSONArray(key);
+
+                        final ArrayList<RespokeGroupMessage> messageList =
+                                new ArrayList<>(jsonMessages.length());
+
+                        for (int i = 0; i < jsonMessages.length(); i++) {
+                            final JSONObject jsonMessage = jsonMessages.getJSONObject(i);
+                            final RespokeGroupMessage message = buildGroupMessage(jsonMessage);
+                            messageList.add(message);
+                        }
+
+                        results.put(key, messageList);
+                    } catch (JSONException e) {
+                        getGroupHistoriesError(completionListener, "Error parsing JSON response");
+                        return;
+                    }
+                }
+
+                new Handler(Looper.getMainLooper()).post(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (completionListener != null) {
+                            completionListener.onSuccess(results);
+                        }
+                    }
+                });
+            }
+
+            @Override
+            public void onError(final String errorMessage) {
+                getGroupHistoriesError(completionListener, errorMessage);
+            }
+        });
+    }
+
```

Comment:
This is good, but we also need the API to fetch messages back to a given timestamp.